From ff69b9518d94f4133130adb788caf466e0c562bd Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke@hauke-m.de>
Date: Tue, 10 May 2022 18:32:52 +0000
Subject: [PATCH 1/4] kernel: bump 5.10 to 5.10.116

Removed upstreamed:
  generic/backport-5.10/900-regulator-consumer-Add-missing-stubs-to-regulator-co.patch

All other patches automatically rebased.

Compile-tested: lantiq/xrx200, armvirt/64
Run-tested: lantiq/xrx200, armvirt/64

Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
---
 include/kernel-5.10                           |  4 +-
 ...er-Add-missing-stubs-to-regulator-co.patch | 79 -------------------
 .../generic/hack-5.10/902-debloat_proc.patch  |  2 +-
 ...-add-support-for-Toshiba-TC58NVG0S3H.patch |  7 +-
 ...Mangle-bootloader-s-kernel-arguments.patch |  2 +-
 .../900-arm-add-cmdline-override.patch        |  2 +-
 ...Mangle-bootloader-s-kernel-arguments.patch |  2 +-
 ...Mangle-bootloader-s-kernel-arguments.patch |  2 +-
 8 files changed, 8 insertions(+), 92 deletions(-)
 delete mode 100644 target/linux/generic/backport-5.10/900-regulator-consumer-Add-missing-stubs-to-regulator-co.patch

diff --git a/include/kernel-5.10 b/include/kernel-5.10
index 405640f695c0..348ba6152a06 100644
--- a/include/kernel-5.10
+++ b/include/kernel-5.10
@@ -1,2 +1,2 @@
-LINUX_VERSION-5.10 = .115
-LINUX_KERNEL_HASH-5.10.115 = f5e417b32f89318b6d0a230109a592ffd68997817463dc4692fa49ec7fe42f71
+LINUX_VERSION-5.10 = .116
+LINUX_KERNEL_HASH-5.10.116 = 806692ba7436f9cfbc32709fe20097dbde4ae5e2f4ae582f39b1d91ce5d20b49
diff --git a/target/linux/generic/backport-5.10/900-regulator-consumer-Add-missing-stubs-to-regulator-co.patch b/target/linux/generic/backport-5.10/900-regulator-consumer-Add-missing-stubs-to-regulator-co.patch
deleted file mode 100644
index c5470e8b82cb..000000000000
--- a/target/linux/generic/backport-5.10/900-regulator-consumer-Add-missing-stubs-to-regulator-co.patch
+++ /dev/null
@@ -1,79 +0,0 @@
-From a71658c7db0b2be9fcab3522aeabe11792a4f481 Mon Sep 17 00:00:00 2001
-From: Dmitry Osipenko <digetx@gmail.com>
-Date: Wed, 20 Jan 2021 23:58:44 +0300
-Subject: [PATCH] regulator: consumer: Add missing stubs to
- regulator/consumer.h
-MIME-Version: 1.0
-Content-Type: text/plain; charset=UTF-8
-Content-Transfer-Encoding: 8bit
-
-commit 51dfb6ca3728bd0a0a3c23776a12d2a15a1d2457 upstream.
-
-Add missing stubs to regulator/consumer.h in order to fix COMPILE_TEST
-of the kernel. In particular this should fix compile-testing of OPP core
-because of a missing stub for regulator_sync_voltage().
-
-Reported-by: kernel test robot <lkp@intel.com>
-Signed-off-by: Dmitry Osipenko <digetx@gmail.com>
-Link: https://lore.kernel.org/r/20210120205844.12658-1-digetx@gmail.com
-Signed-off-by: Mark Brown <broonie@kernel.org>
-Cc: Bjørn Mork <bjorn@mork.no>
-Cc: Sudip Mukherjee <sudipm.mukherjee@gmail.com>
-Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
----
- include/linux/regulator/consumer.h | 30 ++++++++++++++++++++++++++++++
- 1 file changed, 30 insertions(+)
-
---- a/include/linux/regulator/consumer.h
-+++ b/include/linux/regulator/consumer.h
-@@ -332,6 +332,12 @@ regulator_get_exclusive(struct device *d
- }
- 
- static inline struct regulator *__must_check
-+devm_regulator_get_exclusive(struct device *dev, const char *id)
-+{
-+	return ERR_PTR(-ENODEV);
-+}
-+
-+static inline struct regulator *__must_check
- regulator_get_optional(struct device *dev, const char *id)
- {
- 	return ERR_PTR(-ENODEV);
-@@ -486,6 +492,11 @@ static inline int regulator_get_voltage(
- 	return -EINVAL;
- }
- 
-+static inline int regulator_sync_voltage(struct regulator *regulator)
-+{
-+	return -EINVAL;
-+}
-+
- static inline int regulator_is_supported_voltage(struct regulator *regulator,
- 				   int min_uV, int max_uV)
- {
-@@ -578,6 +589,25 @@ static inline int devm_regulator_unregis
- 	return 0;
- }
- 
-+static inline int regulator_suspend_enable(struct regulator_dev *rdev,
-+					   suspend_state_t state)
-+{
-+	return -EINVAL;
-+}
-+
-+static inline int regulator_suspend_disable(struct regulator_dev *rdev,
-+					    suspend_state_t state)
-+{
-+	return -EINVAL;
-+}
-+
-+static inline int regulator_set_suspend_voltage(struct regulator *regulator,
-+						int min_uV, int max_uV,
-+						suspend_state_t state)
-+{
-+	return -EINVAL;
-+}
-+
- static inline void *regulator_get_drvdata(struct regulator *regulator)
- {
- 	return NULL;
diff --git a/target/linux/generic/hack-5.10/902-debloat_proc.patch b/target/linux/generic/hack-5.10/902-debloat_proc.patch
index 33a3e3c5d5d1..0838e2c43c32 100644
--- a/target/linux/generic/hack-5.10/902-debloat_proc.patch
+++ b/target/linux/generic/hack-5.10/902-debloat_proc.patch
@@ -246,7 +246,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  				&vmalloc_op,
 --- a/mm/vmstat.c
 +++ b/mm/vmstat.c
-@@ -2044,10 +2044,12 @@ void __init init_mm_internals(void)
+@@ -2040,10 +2040,12 @@ void __init init_mm_internals(void)
  	start_shepherd_timer();
  #endif
  #ifdef CONFIG_PROC_FS
diff --git a/target/linux/generic/pending-5.10/444-mtd-nand-rawnand-add-support-for-Toshiba-TC58NVG0S3H.patch b/target/linux/generic/pending-5.10/444-mtd-nand-rawnand-add-support-for-Toshiba-TC58NVG0S3H.patch
index 54df82887532..27c863b992df 100644
--- a/target/linux/generic/pending-5.10/444-mtd-nand-rawnand-add-support-for-Toshiba-TC58NVG0S3H.patch
+++ b/target/linux/generic/pending-5.10/444-mtd-nand-rawnand-add-support-for-Toshiba-TC58NVG0S3H.patch
@@ -17,11 +17,9 @@ Signed-off-by: Andreas Böhler <dev@aboehler.at>
  drivers/mtd/nand/raw/nand_ids.c | 3 +++
  1 file changed, 3 insertions(+)
 
-diff --git a/drivers/mtd/nand/raw/nand_ids.c b/drivers/mtd/nand/raw/nand_ids.c
-index 6e41902be35f..d64adbd1ce6b 100644
 --- a/drivers/mtd/nand/raw/nand_ids.c
 +++ b/drivers/mtd/nand/raw/nand_ids.c
-@@ -29,6 +29,9 @@ struct nand_flash_dev nand_flash_ids[] = {
+@@ -29,6 +29,9 @@ struct nand_flash_dev nand_flash_ids[] =
  	{"TC58NVG0S3E 1G 3.3V 8-bit",
  		{ .id = {0x98, 0xd1, 0x90, 0x15, 0x76, 0x14, 0x01, 0x00} },
  		  SZ_2K, SZ_128, SZ_128K, 0, 8, 64, NAND_ECC_INFO(1, SZ_512), },
@@ -31,6 +29,3 @@ index 6e41902be35f..d64adbd1ce6b 100644
  	{"TC58NVG2S0F 4G 3.3V 8-bit",
  		{ .id = {0x98, 0xdc, 0x90, 0x26, 0x76, 0x15, 0x01, 0x08} },
  		  SZ_4K, SZ_512, SZ_256K, 0, 8, 224, NAND_ECC_INFO(4, SZ_512) },
--- 
-2.35.1
-
diff --git a/target/linux/ipq806x/patches-5.10/0067-generic-Mangle-bootloader-s-kernel-arguments.patch b/target/linux/ipq806x/patches-5.10/0067-generic-Mangle-bootloader-s-kernel-arguments.patch
index 8f669319edbe..e9f2d059371c 100644
--- a/target/linux/ipq806x/patches-5.10/0067-generic-Mangle-bootloader-s-kernel-arguments.patch
+++ b/target/linux/ipq806x/patches-5.10/0067-generic-Mangle-bootloader-s-kernel-arguments.patch
@@ -22,7 +22,7 @@ Signed-off-by: Adrian Panella <ianchi74@outlook.com>
 
 --- a/arch/arm/Kconfig
 +++ b/arch/arm/Kconfig
-@@ -1781,6 +1781,17 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_EXTEN
+@@ -1777,6 +1777,17 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_EXTEN
  	  The command-line arguments provided by the boot loader will be
  	  appended to the the device tree bootargs property.
  
diff --git a/target/linux/ipq806x/patches-5.10/900-arm-add-cmdline-override.patch b/target/linux/ipq806x/patches-5.10/900-arm-add-cmdline-override.patch
index dba917872c0c..ddc3b7cb5279 100644
--- a/target/linux/ipq806x/patches-5.10/900-arm-add-cmdline-override.patch
+++ b/target/linux/ipq806x/patches-5.10/900-arm-add-cmdline-override.patch
@@ -1,6 +1,6 @@
 --- a/arch/arm/Kconfig
 +++ b/arch/arm/Kconfig
-@@ -1794,6 +1794,14 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_MANGL
+@@ -1790,6 +1790,14 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_MANGL
  
  endchoice
  
diff --git a/target/linux/mvebu/patches-5.10/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch b/target/linux/mvebu/patches-5.10/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch
index 36f10fcfb52c..04c80113c152 100644
--- a/target/linux/mvebu/patches-5.10/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch
+++ b/target/linux/mvebu/patches-5.10/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch
@@ -28,7 +28,7 @@ Signed-off-by: Michael Gray <michael.gray@lantisproject.com>
 
 --- a/arch/arm/Kconfig
 +++ b/arch/arm/Kconfig
-@@ -1781,6 +1781,17 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_EXTEN
+@@ -1777,6 +1777,17 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_EXTEN
  	  The command-line arguments provided by the boot loader will be
  	  appended to the the device tree bootargs property.
  
diff --git a/target/linux/oxnas/patches-5.10/996-generic-Mangle-bootloader-s-kernel-arguments.patch b/target/linux/oxnas/patches-5.10/996-generic-Mangle-bootloader-s-kernel-arguments.patch
index 898e60230656..8f56bdbf9c29 100644
--- a/target/linux/oxnas/patches-5.10/996-generic-Mangle-bootloader-s-kernel-arguments.patch
+++ b/target/linux/oxnas/patches-5.10/996-generic-Mangle-bootloader-s-kernel-arguments.patch
@@ -22,7 +22,7 @@ Signed-off-by: Adrian Panella <ianchi74@outlook.com>
 
 --- a/arch/arm/Kconfig
 +++ b/arch/arm/Kconfig
-@@ -1781,6 +1781,17 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_EXTEN
+@@ -1777,6 +1777,17 @@ config ARM_ATAG_DTB_COMPAT_CMDLINE_EXTEN
  	  The command-line arguments provided by the boot loader will be
  	  appended to the the device tree bootargs property.
  

From 23a90b8ed9c58b979577acb5f81bfa280b06a508 Mon Sep 17 00:00:00 2001
From: John Audia <therealgraysky@proton.me>
Date: Sun, 22 May 2022 07:12:09 -0400
Subject: [PATCH 2/4] kernel: bump 5.10 to 5.10.117

Removed upstreamed:
    backport-5.10/890-v5.19-net-sfp-Add-tx-fault-workaround-for-Huawei-MA5671A-SFP-ON.patch

All patches automatically rebased.

Build system: x86_64
Build-tested: ipq806x/R7800

Signed-off-by: John Audia <therealgraysky@proton.me>
---
 include/kernel-5.10                           |  4 +-
 ...-support-BCM4908-s-integrated-switch.patch | 10 ++--
 ..._sf2-setup-BCM4908-internal-crossbar.patch | 10 ++--
 ...-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch |  2 +-
 ...-add-function-finding-RGMII-register.patch |  6 +-
 ...-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch |  2 +-
 ...Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch |  4 +-
 ...dsa-bcm_sf2-refactor-LED-regs-access.patch |  2 +-
 ...m_sf2-enable-GPHY-for-switch-probing.patch |  2 +-
 ...sf2-keep-GPHY-enabled-on-the-BCM4908.patch |  2 +-
 ...resolve_btfids-Build-with-host-flags.patch |  7 +--
 ...the-dst-buffer-to-of_get_mac_address.patch |  4 +-
 ...workaround-for-Huawei-MA5671A-SFP-ON.patch | 57 -------------------
 ...y-simplify-phy_link_change-arguments.patch |  8 +--
 .../721-NET-no-auto-carrier-off-support.patch |  4 +-
 15 files changed, 31 insertions(+), 93 deletions(-)
 delete mode 100644 target/linux/generic/backport-5.10/890-v5.19-net-sfp-Add-tx-fault-workaround-for-Huawei-MA5671A-SFP-ON.patch

diff --git a/include/kernel-5.10 b/include/kernel-5.10
index 348ba6152a06..cfc5c9205586 100644
--- a/include/kernel-5.10
+++ b/include/kernel-5.10
@@ -1,2 +1,2 @@
-LINUX_VERSION-5.10 = .116
-LINUX_KERNEL_HASH-5.10.116 = 806692ba7436f9cfbc32709fe20097dbde4ae5e2f4ae582f39b1d91ce5d20b49
+LINUX_VERSION-5.10 = .117
+LINUX_KERNEL_HASH-5.10.117 = 703ef7112cdf889950dd864ba482d04bd87d83d3df3a98ab8d4ba8a7ab1ddcc7
diff --git a/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch b/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch
index 35017bf1e434..5af1f3da32ed 100644
--- a/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch
+++ b/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch
@@ -68,8 +68,8 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  			offset = CORE_STS_OVERRIDE_IMP;
  		else
  			offset = CORE_STS_OVERRIDE_IMP2;
-@@ -711,7 +712,8 @@ static void bcm_sf2_sw_mac_link_down(str
- 	u32 reg, offset;
+@@ -714,7 +715,8 @@ static void bcm_sf2_sw_mac_link_down(str
+ 		return;
  
  	if (port != core_readl(priv, CORE_IMP0_PRT_ID)) {
 -		if (priv->type == BCM7445_DEVICE_ID)
@@ -78,7 +78,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
  		else
  			offset = CORE_STS_OVERRIDE_GMIIP2_PORT(port);
-@@ -738,7 +740,8 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -741,7 +743,8 @@ static void bcm_sf2_sw_mac_link_up(struc
  	bcm_sf2_sw_mac_link_set(ds, port, interface, true);
  
  	if (port != core_readl(priv, CORE_IMP0_PRT_ID)) {
@@ -88,7 +88,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
  		else
  			offset = CORE_STS_OVERRIDE_GMIIP2_PORT(port);
-@@ -1131,6 +1134,30 @@ struct bcm_sf2_of_data {
+@@ -1134,6 +1137,30 @@ struct bcm_sf2_of_data {
  	unsigned int num_cfp_rules;
  };
  
@@ -119,7 +119,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  /* Register offsets for the SWITCH_REG_* block */
  static const u16 bcm_sf2_7445_reg_offsets[] = {
  	[REG_SWITCH_CNTRL]	= 0x00,
-@@ -1179,6 +1206,9 @@ static const struct bcm_sf2_of_data bcm_
+@@ -1182,6 +1209,9 @@ static const struct bcm_sf2_of_data bcm_
  };
  
  static const struct of_device_id bcm_sf2_of_match[] = {
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch
index b9ec7af1ae4c..6d76640ead2e 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch
@@ -82,7 +82,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static void bcm_sf2_intr_disable(struct bcm_sf2_priv *priv)
  {
  	intrl2_0_mask_set(priv, 0xffffffff);
-@@ -872,6 +910,8 @@ static int bcm_sf2_sw_resume(struct dsa_
+@@ -875,6 +913,8 @@ static int bcm_sf2_sw_resume(struct dsa_
  		return ret;
  	}
  
@@ -91,7 +91,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	ret = bcm_sf2_cfp_resume(ds);
  	if (ret)
  		return ret;
-@@ -1143,6 +1183,7 @@ struct bcm_sf2_of_data {
+@@ -1146,6 +1186,7 @@ struct bcm_sf2_of_data {
  	const u16 *reg_offsets;
  	unsigned int core_reg_align;
  	unsigned int num_cfp_rules;
@@ -99,7 +99,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  };
  
  static const u16 bcm_sf2_4908_reg_offsets[] = {
-@@ -1167,6 +1208,7 @@ static const struct bcm_sf2_of_data bcm_
+@@ -1170,6 +1211,7 @@ static const struct bcm_sf2_of_data bcm_
  	.core_reg_align	= 0,
  	.reg_offsets	= bcm_sf2_4908_reg_offsets,
  	.num_cfp_rules	= 0, /* FIXME */
@@ -107,7 +107,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  };
  
  /* Register offsets for the SWITCH_REG_* block */
-@@ -1277,6 +1319,7 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1280,6 +1322,7 @@ static int bcm_sf2_sw_probe(struct platf
  	priv->reg_offsets = data->reg_offsets;
  	priv->core_reg_align = data->core_reg_align;
  	priv->num_cfp_rules = data->num_cfp_rules;
@@ -115,7 +115,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  
  	priv->rcdev = devm_reset_control_get_optional_exclusive(&pdev->dev,
  								"switch");
-@@ -1350,6 +1393,8 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1353,6 +1396,8 @@ static int bcm_sf2_sw_probe(struct platf
  		goto out_clk_mdiv;
  	}
  
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch
index b99bf010bd65..9ff1051f352c 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch
@@ -14,7 +14,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -1207,7 +1207,7 @@ static const struct bcm_sf2_of_data bcm_
+@@ -1210,7 +1210,7 @@ static const struct bcm_sf2_of_data bcm_
  	.type		= BCM4908_DEVICE_ID,
  	.core_reg_align	= 0,
  	.reg_offsets	= bcm_sf2_4908_reg_offsets,
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch
index 132a734ac4ef..281a8bbcda9f 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch
@@ -113,7 +113,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  }
  
  static void bcm_sf2_sw_mac_link_down(struct dsa_switch *ds, int port,
-@@ -784,11 +815,15 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -787,11 +818,15 @@ static void bcm_sf2_sw_mac_link_up(struc
  {
  	struct bcm_sf2_priv *priv = bcm_sf2_to_priv(ds);
  	struct ethtool_eee *p = &priv->dev->ports[port].eee;
@@ -130,7 +130,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		if (priv->type == BCM4908_DEVICE_ID ||
  		    priv->type == BCM7445_DEVICE_ID)
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
-@@ -799,7 +834,7 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -802,7 +837,7 @@ static void bcm_sf2_sw_mac_link_up(struc
  		    interface == PHY_INTERFACE_MODE_RGMII_TXID ||
  		    interface == PHY_INTERFACE_MODE_MII ||
  		    interface == PHY_INTERFACE_MODE_REVMII) {
@@ -139,7 +139,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  			reg &= ~(RX_PAUSE_EN | TX_PAUSE_EN);
  
  			if (tx_pause)
-@@ -807,7 +842,7 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -810,7 +845,7 @@ static void bcm_sf2_sw_mac_link_up(struc
  			if (rx_pause)
  				reg |= RX_PAUSE_EN;
  
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch
index 67c633a930bb..312f46723f9c 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch
@@ -33,7 +33,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		break;
  	default:
  		switch (port) {
-@@ -1230,9 +1235,7 @@ static const u16 bcm_sf2_4908_reg_offset
+@@ -1233,9 +1238,7 @@ static const u16 bcm_sf2_4908_reg_offset
  	[REG_PHY_REVISION]	= 0x14,
  	[REG_SPHY_CNTRL]	= 0x24,
  	[REG_CROSSBAR]		= 0xc8,
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch
index 5e5d6d939d89..f7e9bcb5a259 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch
@@ -23,7 +23,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -824,11 +824,9 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -827,11 +827,9 @@ static void bcm_sf2_sw_mac_link_up(struc
  	bcm_sf2_sw_mac_link_set(ds, port, interface, true);
  
  	if (port != core_readl(priv, CORE_IMP0_PRT_ID)) {
@@ -36,7 +36,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		if (priv->type == BCM4908_DEVICE_ID ||
  		    priv->type == BCM7445_DEVICE_ID)
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
-@@ -839,6 +837,7 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -842,6 +840,7 @@ static void bcm_sf2_sw_mac_link_up(struc
  		    interface == PHY_INTERFACE_MODE_RGMII_TXID ||
  		    interface == PHY_INTERFACE_MODE_MII ||
  		    interface == PHY_INTERFACE_MODE_REVMII) {
diff --git a/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch b/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch
index bf45cdcf24be..e31d93b39633 100644
--- a/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch
+++ b/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch
@@ -82,7 +82,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	}
  }
  
-@@ -1235,9 +1272,14 @@ static const u16 bcm_sf2_4908_reg_offset
+@@ -1238,9 +1275,14 @@ static const u16 bcm_sf2_4908_reg_offset
  	[REG_SPHY_CNTRL]	= 0x24,
  	[REG_CROSSBAR]		= 0xc8,
  	[REG_RGMII_11_CNTRL]	= 0x014c,
diff --git a/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch b/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch
index a53fca0e9036..4aec8f843991 100644
--- a/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch
+++ b/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch
@@ -29,7 +29,7 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -1532,10 +1532,14 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1535,10 +1535,14 @@ static int bcm_sf2_sw_probe(struct platf
  	rev = reg_readl(priv, REG_PHY_REVISION);
  	priv->hw_params.gphy_rev = rev & PHY_REVISION_MASK;
  
diff --git a/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch b/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch
index 00ffb19f6fcf..78f6e9042792 100644
--- a/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch
+++ b/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch
@@ -15,7 +15,7 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -1546,6 +1546,12 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1549,6 +1549,12 @@ static int bcm_sf2_sw_probe(struct platf
  		 priv->hw_params.core_rev >> 8, priv->hw_params.core_rev & 0xff,
  		 priv->irq0, priv->irq1);
  
diff --git a/target/linux/generic/backport-5.10/200-v5.18-tools-resolve_btfids-Build-with-host-flags.patch b/target/linux/generic/backport-5.10/200-v5.18-tools-resolve_btfids-Build-with-host-flags.patch
index 489575fb3526..caec8db5d692 100644
--- a/target/linux/generic/backport-5.10/200-v5.18-tools-resolve_btfids-Build-with-host-flags.patch
+++ b/target/linux/generic/backport-5.10/200-v5.18-tools-resolve_btfids-Build-with-host-flags.patch
@@ -24,8 +24,6 @@ Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
  tools/bpf/resolve_btfids/Makefile | 6 ++++--
  1 file changed, 4 insertions(+), 2 deletions(-)
 
-diff --git a/tools/bpf/resolve_btfids/Makefile b/tools/bpf/resolve_btfids/Makefile
-index af9f9d3534c9..e727058bfe15 100644
 --- a/tools/bpf/resolve_btfids/Makefile
 +++ b/tools/bpf/resolve_btfids/Makefile
 @@ -23,6 +23,8 @@ CC       = $(HOSTCC)
@@ -37,7 +35,7 @@ index af9f9d3534c9..e727058bfe15 100644
  
  OUTPUT ?= $(srctree)/tools/bpf/resolve_btfids/
  
-@@ -45,9 +47,9 @@ $(SUBCMDOBJ): fixdep FORCE | $(OUTPUT)/libsubcmd
+@@ -45,9 +47,9 @@ $(SUBCMDOBJ): fixdep FORCE | $(OUTPUT)/l
  	$(Q)$(MAKE) -C $(SUBCMD_SRC) OUTPUT=$(abspath $(dir $@))/ $(abspath $@)
  
  $(BPFOBJ): $(wildcard $(LIBBPF_SRC)/*.[ch] $(LIBBPF_SRC)/Makefile) | $(OUTPUT)/libbpf
@@ -49,6 +47,3 @@ index af9f9d3534c9..e727058bfe15 100644
            -I$(srctree)/tools/include \
            -I$(srctree)/tools/include/uapi \
            -I$(LIBBPF_SRC) \
--- 
-2.35.1
-
diff --git a/target/linux/generic/backport-5.10/732-net-next-1-of-net-pass-the-dst-buffer-to-of_get_mac_address.patch b/target/linux/generic/backport-5.10/732-net-next-1-of-net-pass-the-dst-buffer-to-of_get_mac_address.patch
index a4a9f5624ec9..3b59e4532c2e 100644
--- a/target/linux/generic/backport-5.10/732-net-next-1-of-net-pass-the-dst-buffer-to-of_get_mac_address.patch
+++ b/target/linux/generic/backport-5.10/732-net-next-1-of-net-pass-the-dst-buffer-to-of_get_mac_address.patch
@@ -1641,7 +1641,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	lp->coalesce_count_tx = XAXIDMA_DFT_TX_THRESHOLD;
 --- a/drivers/net/ethernet/xilinx/xilinx_emaclite.c
 +++ b/drivers/net/ethernet/xilinx/xilinx_emaclite.c
-@@ -1122,7 +1122,6 @@ static int xemaclite_of_probe(struct pla
+@@ -1107,7 +1107,6 @@ static int xemaclite_of_probe(struct pla
  	struct net_device *ndev = NULL;
  	struct net_local *lp = NULL;
  	struct device *dev = &ofdev->dev;
@@ -1649,7 +1649,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  
  	int rc = 0;
  
-@@ -1164,12 +1163,9 @@ static int xemaclite_of_probe(struct pla
+@@ -1149,12 +1148,9 @@ static int xemaclite_of_probe(struct pla
  	lp->next_rx_buf_to_use = 0x0;
  	lp->tx_ping_pong = get_bool(ofdev, "xlnx,tx-ping-pong");
  	lp->rx_ping_pong = get_bool(ofdev, "xlnx,rx-ping-pong");
diff --git a/target/linux/generic/backport-5.10/890-v5.19-net-sfp-Add-tx-fault-workaround-for-Huawei-MA5671A-SFP-ON.patch b/target/linux/generic/backport-5.10/890-v5.19-net-sfp-Add-tx-fault-workaround-for-Huawei-MA5671A-SFP-ON.patch
deleted file mode 100644
index 7a9de8ee1e05..000000000000
--- a/target/linux/generic/backport-5.10/890-v5.19-net-sfp-Add-tx-fault-workaround-for-Huawei-MA5671A-SFP-ON.patch
+++ /dev/null
@@ -1,57 +0,0 @@
-From f81d97cb646ab8b90fb181d66fccaf9589990de6 Mon Sep 17 00:00:00 2001
-From: Matthew Hagan <mnhagan88@gmail.com>
-Date: Sat, 30 Apr 2022 11:00:49 +0100
-Subject: [PATCH v2] net: sfp: Add tx-fault workaround for Huawei MA5671A SFP
- ONT
-
-As noted elsewhere, various GPON SFP modules exhibit non-standard
-TX-fault behaviour. In the tested case, the Huawei MA5671A, when used
-in combination with a Marvell mv88e6085 switch, was found to
-persistently assert TX-fault, resulting in the module being disabled.
-
-This patch adds a quirk to ignore the SFP_F_TX_FAULT state, allowing the
-module to function.
-
-Change from v1: removal of erroneous return statment (Andrew Lunn)
-
-Signed-off-by: Matthew Hagan <mnhagan88@gmail.com>
-Reviewed-by: Andrew Lunn <andrew@lunn.ch>
----
- drivers/net/phy/sfp.c | 12 +++++++++++-
- 1 file changed, 11 insertions(+), 1 deletion(-)
-
---- a/drivers/net/phy/sfp.c
-+++ b/drivers/net/phy/sfp.c
-@@ -249,6 +249,7 @@ struct sfp {
- 	struct sfp_eeprom_id id;
- 	unsigned int module_power_mW;
- 	unsigned int module_t_start_up;
-+	bool tx_fault_ignore;
- 
- #if IS_ENABLED(CONFIG_HWMON)
- 	struct sfp_diag diag;
-@@ -1893,6 +1894,12 @@ static int sfp_sm_mod_probe(struct sfp *
- 	else
- 		sfp->module_t_start_up = T_START_UP;
- 
-+	if (!memcmp(id.base.vendor_name, "HUAWEI          ", 16) &&
-+	    !memcmp(id.base.vendor_pn, "MA5671A         ", 16))
-+		sfp->tx_fault_ignore = true;
-+	else
-+		sfp->tx_fault_ignore = false;
-+
- 	return 0;
- }
- 
-@@ -2320,7 +2327,10 @@ static void sfp_check_state(struct sfp *
- 	mutex_lock(&sfp->st_mutex);
- 	state = sfp_get_state(sfp);
- 	changed = state ^ sfp->state;
--	changed &= SFP_F_PRESENT | SFP_F_LOS | SFP_F_TX_FAULT;
-+	if (sfp->tx_fault_ignore)
-+		changed &= SFP_F_PRESENT | SFP_F_LOS;
-+	else
-+		changed &= SFP_F_PRESENT | SFP_F_LOS | SFP_F_TX_FAULT;
- 
- 	for (i = 0; i < GPIO_MAX; i++)
- 		if (changed & BIT(i))
diff --git a/target/linux/ramips/patches-5.10/720-Revert-net-phy-simplify-phy_link_change-arguments.patch b/target/linux/ramips/patches-5.10/720-Revert-net-phy-simplify-phy_link_change-arguments.patch
index a73d5e7b1eb7..6e8b48829018 100644
--- a/target/linux/ramips/patches-5.10/720-Revert-net-phy-simplify-phy_link_change-arguments.patch
+++ b/target/linux/ramips/patches-5.10/720-Revert-net-phy-simplify-phy_link_change-arguments.patch
@@ -33,7 +33,7 @@ still required by target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c
  	phy_led_trigger_change_speed(phydev);
  }
  
-@@ -567,7 +567,7 @@ int phy_start_cable_test(struct phy_devi
+@@ -572,7 +572,7 @@ int phy_start_cable_test(struct phy_devi
  		goto out;
  
  	/* Mark the carrier down until the test is complete */
@@ -42,7 +42,7 @@ still required by target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c
  
  	netif_testing_on(dev);
  	err = phydev->drv->cable_test_start(phydev);
-@@ -638,7 +638,7 @@ int phy_start_cable_test_tdr(struct phy_
+@@ -643,7 +643,7 @@ int phy_start_cable_test_tdr(struct phy_
  		goto out;
  
  	/* Mark the carrier down until the test is complete */
@@ -51,7 +51,7 @@ still required by target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c
  
  	netif_testing_on(dev);
  	err = phydev->drv->cable_test_tdr_start(phydev, config);
-@@ -709,7 +709,7 @@ static int phy_check_link_status(struct
+@@ -714,7 +714,7 @@ static int phy_check_link_status(struct
  		phy_link_up(phydev);
  	} else if (!phydev->link && phydev->state != PHY_NOLINK) {
  		phydev->state = PHY_NOLINK;
@@ -60,7 +60,7 @@ still required by target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c
  	}
  
  	return 0;
-@@ -1191,7 +1191,7 @@ void phy_state_machine(struct work_struc
+@@ -1226,7 +1226,7 @@ void phy_state_machine(struct work_struc
  	case PHY_HALTED:
  		if (phydev->link) {
  			phydev->link = 0;
diff --git a/target/linux/ramips/patches-5.10/721-NET-no-auto-carrier-off-support.patch b/target/linux/ramips/patches-5.10/721-NET-no-auto-carrier-off-support.patch
index b83daffd4f45..5e82c13d2a23 100644
--- a/target/linux/ramips/patches-5.10/721-NET-no-auto-carrier-off-support.patch
+++ b/target/linux/ramips/patches-5.10/721-NET-no-auto-carrier-off-support.patch
@@ -11,7 +11,7 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
 
 --- a/drivers/net/phy/phy.c
 +++ b/drivers/net/phy/phy.c
-@@ -709,7 +709,10 @@ static int phy_check_link_status(struct
+@@ -714,7 +714,10 @@ static int phy_check_link_status(struct
  		phy_link_up(phydev);
  	} else if (!phydev->link && phydev->state != PHY_NOLINK) {
  		phydev->state = PHY_NOLINK;
@@ -23,7 +23,7 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
  	}
  
  	return 0;
-@@ -1191,7 +1194,10 @@ void phy_state_machine(struct work_struc
+@@ -1226,7 +1229,10 @@ void phy_state_machine(struct work_struc
  	case PHY_HALTED:
  		if (phydev->link) {
  			phydev->link = 0;

From f1dec0c454d678e3741daf74580327f416ed6401 Mon Sep 17 00:00:00 2001
From: John Audia <therealgraysky@proton.me>
Date: Wed, 25 May 2022 13:09:23 -0400
Subject: [PATCH 3/4] kernel: bump 5.10 to 5.10.118

Removed upstreamed:
    generic-backport/774-v5.15-1-igc-remove-_I_PHY_ID-checking.patch
    generic-backport/774-v5.15-2-igc-remove-phy-type-checking.patch

All patches automatically rebased.

Build system: x86_64
Build-tested: ipq806x/R7800

Signed-off-by: John Audia <therealgraysky@proton.me>
---
 include/kernel-5.10                           |  4 +-
 ...at91-re-factor-clocks-suspend-resume.patch |  6 +-
 ...5.15-1-igc-remove-_I_PHY_ID-checking.patch | 73 -------------------
 ...v5.15-2-igc-remove-phy-type-checking.patch | 43 -----------
 .../generic/hack-5.10/204-module_strip.patch  |  4 +-
 .../hack-5.10/600-bridge_offload.patch        |  4 +-
 ...ge_allow_receiption_on_disabled_port.patch |  4 +-
 ...d-knob-for-filtering-rx-tx-BPDU-pack.patch |  2 +-
 ...PS-lantiq-improve-USB-initialization.patch |  2 +-
 ...add-gphy-clk-src-device-tree-binding.patch |  4 +-
 10 files changed, 15 insertions(+), 131 deletions(-)
 delete mode 100644 target/linux/generic/backport-5.10/774-v5.15-1-igc-remove-_I_PHY_ID-checking.patch
 delete mode 100644 target/linux/generic/backport-5.10/774-v5.15-2-igc-remove-phy-type-checking.patch

diff --git a/include/kernel-5.10 b/include/kernel-5.10
index cfc5c9205586..c05a5d842523 100644
--- a/include/kernel-5.10
+++ b/include/kernel-5.10
@@ -1,2 +1,2 @@
-LINUX_VERSION-5.10 = .117
-LINUX_KERNEL_HASH-5.10.117 = 703ef7112cdf889950dd864ba482d04bd87d83d3df3a98ab8d4ba8a7ab1ddcc7
+LINUX_VERSION-5.10 = .118
+LINUX_KERNEL_HASH-5.10.118 = c5e020f187a195ab0924fe7e71b8ab8fb92936ab47bc9d15e78daabddefe0b4b
diff --git a/target/linux/at91/patches-5.10/234-clk-at91-re-factor-clocks-suspend-resume.patch b/target/linux/at91/patches-5.10/234-clk-at91-re-factor-clocks-suspend-resume.patch
index fc08b1fce6a6..5d399f6535e8 100644
--- a/target/linux/at91/patches-5.10/234-clk-at91-re-factor-clocks-suspend-resume.patch
+++ b/target/linux/at91/patches-5.10/234-clk-at91-re-factor-clocks-suspend-resume.patch
@@ -93,7 +93,7 @@ Signed-off-by: Stephen Boyd <sboyd@kernel.org>
  	return 0;
  }
  
-@@ -245,6 +256,23 @@ static int clk_generated_set_rate(struct
+@@ -249,6 +260,23 @@ static int clk_generated_set_rate(struct
  	return 0;
  }
  
@@ -117,7 +117,7 @@ Signed-off-by: Stephen Boyd <sboyd@kernel.org>
  static const struct clk_ops generated_ops = {
  	.enable = clk_generated_enable,
  	.disable = clk_generated_disable,
-@@ -254,6 +282,8 @@ static const struct clk_ops generated_op
+@@ -258,6 +286,8 @@ static const struct clk_ops generated_op
  	.get_parent = clk_generated_get_parent,
  	.set_parent = clk_generated_set_parent,
  	.set_rate = clk_generated_set_rate,
@@ -126,7 +126,7 @@ Signed-off-by: Stephen Boyd <sboyd@kernel.org>
  };
  
  /**
-@@ -320,8 +350,6 @@ at91_clk_register_generated(struct regma
+@@ -324,8 +354,6 @@ at91_clk_register_generated(struct regma
  	if (ret) {
  		kfree(gck);
  		hw = ERR_PTR(ret);
diff --git a/target/linux/generic/backport-5.10/774-v5.15-1-igc-remove-_I_PHY_ID-checking.patch b/target/linux/generic/backport-5.10/774-v5.15-1-igc-remove-_I_PHY_ID-checking.patch
deleted file mode 100644
index d7df0685dd2d..000000000000
--- a/target/linux/generic/backport-5.10/774-v5.15-1-igc-remove-_I_PHY_ID-checking.patch
+++ /dev/null
@@ -1,73 +0,0 @@
-From 7c496de538eebd8212dc2a3c9a468386b264d0d4 Mon Sep 17 00:00:00 2001
-From: Sasha Neftin <sasha.neftin@intel.com>
-Date: Wed, 7 Jul 2021 08:14:40 +0300
-Subject: igc: Remove _I_PHY_ID checking
-
-i225 devices have only one PHY vendor. There is no point checking
-_I_PHY_ID during the link establishment and auto-negotiation process.
-This patch comes to clean up these pointless checkings.
-
-Signed-off-by: Sasha Neftin <sasha.neftin@intel.com>
-Tested-by: Dvora Fuxbrumer <dvorax.fuxbrumer@linux.intel.com>
-Signed-off-by: Tony Nguyen <anthony.l.nguyen@intel.com>
----
- drivers/net/ethernet/intel/igc/igc_base.c | 10 +---------
- drivers/net/ethernet/intel/igc/igc_main.c |  3 +--
- drivers/net/ethernet/intel/igc/igc_phy.c  |  6 ++----
- 3 files changed, 4 insertions(+), 15 deletions(-)
-
-(limited to 'drivers/net/ethernet/intel/igc')
-
---- a/drivers/net/ethernet/intel/igc/igc_base.c
-+++ b/drivers/net/ethernet/intel/igc/igc_base.c
-@@ -187,15 +187,7 @@ static s32 igc_init_phy_params_base(stru
- 
- 	igc_check_for_copper_link(hw);
- 
--	/* Verify phy id and set remaining function pointers */
--	switch (phy->id) {
--	case I225_I_PHY_ID:
--		phy->type	= igc_phy_i225;
--		break;
--	default:
--		ret_val = -IGC_ERR_PHY;
--		goto out;
--	}
-+	phy->type = igc_phy_i225;
- 
- out:
- 	return ret_val;
---- a/drivers/net/ethernet/intel/igc/igc_main.c
-+++ b/drivers/net/ethernet/intel/igc/igc_main.c
-@@ -4189,8 +4189,7 @@ bool igc_has_link(struct igc_adapter *ad
- 		break;
- 	}
- 
--	if (hw->mac.type == igc_i225 &&
--	    hw->phy.id == I225_I_PHY_ID) {
-+	if (hw->mac.type == igc_i225) {
- 		if (!netif_carrier_ok(adapter->netdev)) {
- 			adapter->flags &= ~IGC_FLAG_NEED_LINK_UPDATE;
- 		} else if (!(adapter->flags & IGC_FLAG_NEED_LINK_UPDATE)) {
---- a/drivers/net/ethernet/intel/igc/igc_phy.c
-+++ b/drivers/net/ethernet/intel/igc/igc_phy.c
-@@ -249,8 +249,7 @@ static s32 igc_phy_setup_autoneg(struct
- 			return ret_val;
- 	}
- 
--	if ((phy->autoneg_mask & ADVERTISE_2500_FULL) &&
--	    hw->phy.id == I225_I_PHY_ID) {
-+	if (phy->autoneg_mask & ADVERTISE_2500_FULL) {
- 		/* Read the MULTI GBT AN Control Register - reg 7.32 */
- 		ret_val = phy->ops.read_reg(hw, (STANDARD_AN_REG_MASK <<
- 					    MMD_DEVADDR_SHIFT) |
-@@ -390,8 +389,7 @@ static s32 igc_phy_setup_autoneg(struct
- 		ret_val = phy->ops.write_reg(hw, PHY_1000T_CTRL,
- 					     mii_1000t_ctrl_reg);
- 
--	if ((phy->autoneg_mask & ADVERTISE_2500_FULL) &&
--	    hw->phy.id == I225_I_PHY_ID)
-+	if (phy->autoneg_mask & ADVERTISE_2500_FULL)
- 		ret_val = phy->ops.write_reg(hw,
- 					     (STANDARD_AN_REG_MASK <<
- 					     MMD_DEVADDR_SHIFT) |
diff --git a/target/linux/generic/backport-5.10/774-v5.15-2-igc-remove-phy-type-checking.patch b/target/linux/generic/backport-5.10/774-v5.15-2-igc-remove-phy-type-checking.patch
deleted file mode 100644
index ad4d1bb0dcad..000000000000
--- a/target/linux/generic/backport-5.10/774-v5.15-2-igc-remove-phy-type-checking.patch
+++ /dev/null
@@ -1,43 +0,0 @@
-From 47bca7de6a4fb8dcb564c7ca14d885c91ed19e03 Mon Sep 17 00:00:00 2001
-From: Sasha Neftin <sasha.neftin@intel.com>
-Date: Sat, 10 Jul 2021 20:57:50 +0300
-Subject: igc: Remove phy->type checking
-
-i225 devices have only one phy->type: copper. There is no point checking
-phy->type during the igc_has_link method from the watchdog that
-invoked every 2 seconds.
-This patch comes to clean up these pointless checkings.
-
-Signed-off-by: Sasha Neftin <sasha.neftin@intel.com>
-Tested-by: Dvora Fuxbrumer <dvorax.fuxbrumer@linux.intel.com>
-Signed-off-by: Tony Nguyen <anthony.l.nguyen@intel.com>
----
- drivers/net/ethernet/intel/igc/igc_main.c | 15 ++++-----------
- 1 file changed, 4 insertions(+), 11 deletions(-)
-
-(limited to 'drivers/net/ethernet/intel/igc')
-
---- a/drivers/net/ethernet/intel/igc/igc_main.c
-+++ b/drivers/net/ethernet/intel/igc/igc_main.c
-@@ -4177,17 +4177,10 @@ bool igc_has_link(struct igc_adapter *ad
- 	 * false until the igc_check_for_link establishes link
- 	 * for copper adapters ONLY
- 	 */
--	switch (hw->phy.media_type) {
--	case igc_media_type_copper:
--		if (!hw->mac.get_link_status)
--			return true;
--		hw->mac.ops.check_for_link(hw);
--		link_active = !hw->mac.get_link_status;
--		break;
--	default:
--	case igc_media_type_unknown:
--		break;
--	}
-+	if (!hw->mac.get_link_status)
-+		return true;
-+	hw->mac.ops.check_for_link(hw);
-+	link_active = !hw->mac.get_link_status;
- 
- 	if (hw->mac.type == igc_i225) {
- 		if (!netif_carrier_ok(adapter->netdev)) {
diff --git a/target/linux/generic/hack-5.10/204-module_strip.patch b/target/linux/generic/hack-5.10/204-module_strip.patch
index 645074821f22..4c6b4454c647 100644
--- a/target/linux/generic/hack-5.10/204-module_strip.patch
+++ b/target/linux/generic/hack-5.10/204-module_strip.patch
@@ -120,7 +120,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  
  #ifdef CONFIG_MODVERSIONS
  
-@@ -3247,9 +3249,11 @@ static int setup_load_info(struct load_i
+@@ -3251,9 +3253,11 @@ static int setup_load_info(struct load_i
  
  static int check_modinfo(struct module *mod, struct load_info *info, int flags)
  {
@@ -133,7 +133,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  	if (flags & MODULE_INIT_IGNORE_VERMAGIC)
  		modmagic = NULL;
  
-@@ -3270,6 +3274,7 @@ static int check_modinfo(struct module *
+@@ -3274,6 +3278,7 @@ static int check_modinfo(struct module *
  				mod->name);
  		add_taint_module(mod, TAINT_OOT_MODULE, LOCKDEP_STILL_OK);
  	}
diff --git a/target/linux/generic/hack-5.10/600-bridge_offload.patch b/target/linux/generic/hack-5.10/600-bridge_offload.patch
index 586353816c2e..b69ea5c7624c 100644
--- a/target/linux/generic/hack-5.10/600-bridge_offload.patch
+++ b/target/linux/generic/hack-5.10/600-bridge_offload.patch
@@ -154,7 +154,7 @@
  
  static int
  br_netif_receive_skb(struct net *net, struct sock *sk, struct sk_buff *skb)
-@@ -162,6 +163,7 @@ int br_handle_frame_finish(struct net *n
+@@ -169,6 +170,7 @@ int br_handle_frame_finish(struct net *n
  			dst->used = now;
  		br_forward(dst->dst, skb, local_rcv, false);
  	} else {
@@ -162,7 +162,7 @@
  		if (!mcast_hit)
  			br_flood(br, skb, pkt_type, local_rcv, false);
  		else
-@@ -280,6 +282,9 @@ static rx_handler_result_t br_handle_fra
+@@ -287,6 +289,9 @@ static rx_handler_result_t br_handle_fra
  	memset(skb->cb, 0, sizeof(struct br_input_skb_cb));
  
  	p = br_port_get_rcu(skb->dev);
diff --git a/target/linux/generic/pending-5.10/150-bridge_allow_receiption_on_disabled_port.patch b/target/linux/generic/pending-5.10/150-bridge_allow_receiption_on_disabled_port.patch
index 6de22b1c7d2d..8ab4b361f097 100644
--- a/target/linux/generic/pending-5.10/150-bridge_allow_receiption_on_disabled_port.patch
+++ b/target/linux/generic/pending-5.10/150-bridge_allow_receiption_on_disabled_port.patch
@@ -15,7 +15,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 
 --- a/net/bridge/br_input.c
 +++ b/net/bridge/br_input.c
-@@ -195,6 +195,9 @@ static void __br_handle_local_finish(str
+@@ -202,6 +202,9 @@ static void __br_handle_local_finish(str
  /* note: already called with rcu_read_lock */
  static int br_handle_local_finish(struct net *net, struct sock *sk, struct sk_buff *skb)
  {
@@ -25,7 +25,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  	__br_handle_local_finish(skb);
  
  	/* return 1 to signal the okfn() was called so it's ok to use the skb */
-@@ -348,6 +351,17 @@ static rx_handler_result_t br_handle_fra
+@@ -355,6 +358,17 @@ static rx_handler_result_t br_handle_fra
  
  forward:
  	switch (p->state) {
diff --git a/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch b/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
index 106f7d356098..d907006fa4dd 100644
--- a/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
+++ b/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
@@ -45,7 +45,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  			if (!(p->flags & BR_BCAST_FLOOD) && skb->dev != br->dev)
 --- a/net/bridge/br_input.c
 +++ b/net/bridge/br_input.c
-@@ -305,6 +305,8 @@ static rx_handler_result_t br_handle_fra
+@@ -312,6 +312,8 @@ static rx_handler_result_t br_handle_fra
  		fwd_mask |= p->group_fwd_mask;
  		switch (dest[5]) {
  		case 0x00:	/* Bridge Group Address */
diff --git a/target/linux/lantiq/patches-5.10/0051-MIPS-lantiq-improve-USB-initialization.patch b/target/linux/lantiq/patches-5.10/0051-MIPS-lantiq-improve-USB-initialization.patch
index 17e07cd63abb..9d62892b566b 100644
--- a/target/linux/lantiq/patches-5.10/0051-MIPS-lantiq-improve-USB-initialization.patch
+++ b/target/linux/lantiq/patches-5.10/0051-MIPS-lantiq-improve-USB-initialization.patch
@@ -41,7 +41,7 @@ Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
  /* the pci enable helper */
  static int pci_enable(struct clk *clk)
  {
-@@ -571,4 +590,5 @@ void __init ltq_soc_init(void)
+@@ -585,4 +604,5 @@ void __init ltq_soc_init(void)
  		clkdev_add_pmu("1e116000.mei", "dfe", 1, 0, PMU_DFE);
  		clkdev_add_pmu("1e100400.serial", NULL, 1, 0, PMU_ASC0);
  	}
diff --git a/target/linux/lantiq/patches-5.10/0301-xrx200-add-gphy-clk-src-device-tree-binding.patch b/target/linux/lantiq/patches-5.10/0301-xrx200-add-gphy-clk-src-device-tree-binding.patch
index 01b87b9d5c74..92c4b564930b 100644
--- a/target/linux/lantiq/patches-5.10/0301-xrx200-add-gphy-clk-src-device-tree-binding.patch
+++ b/target/linux/lantiq/patches-5.10/0301-xrx200-add-gphy-clk-src-device-tree-binding.patch
@@ -1,6 +1,6 @@
 --- a/arch/mips/lantiq/xway/sysctrl.c
 +++ b/arch/mips/lantiq/xway/sysctrl.c
-@@ -426,6 +426,20 @@ static void clkdev_add_clkout(void)
+@@ -440,6 +440,20 @@ static void clkdev_add_clkout(void)
  	}
  }
  
@@ -21,7 +21,7 @@
  /* bring up all register ranges that we need for basic system control */
  void __init ltq_soc_init(void)
  {
-@@ -591,4 +605,6 @@ void __init ltq_soc_init(void)
+@@ -605,4 +619,6 @@ void __init ltq_soc_init(void)
  		clkdev_add_pmu("1e100400.serial", NULL, 1, 0, PMU_ASC0);
  	}
  	usb_set_clock();

From f0bcccc4a54e7512a2901a3f030925e45434ec9c Mon Sep 17 00:00:00 2001
From: John Audia <therealgraysky@proton.me>
Date: Mon, 30 May 2022 08:45:46 -0400
Subject: [PATCH 4/4] kernel: bump 5.10 to 5.10.119

Delete the crypto-lib-blake2s kmod package, as BLAKE2s is now built-in.
Patches automatically rebased.

Build system: x86_64
Build-tested: ipq806x/R7800

Signed-off-by: John Audia <therealgraysky@proton.me>
---
 include/kernel-5.10                           |  4 +-
 package/kernel/linux/modules/crypto.mk        | 42 -------------------
 package/kernel/linux/modules/netsupport.mk    |  1 -
 .../910-unaligned_access_hacks.patch          |  4 +-
 target/linux/generic/config-5.10              |  1 -
 .../hack-5.10/261-lib-arc4-unhide.patch       |  4 +-
 .../205-backtrace_module_info.patch           |  4 +-
 ...hdog-add-realtek-otto-watchdog-timer.patch |  2 +-
 8 files changed, 9 insertions(+), 53 deletions(-)

diff --git a/include/kernel-5.10 b/include/kernel-5.10
index c05a5d842523..f12a386aa517 100644
--- a/include/kernel-5.10
+++ b/include/kernel-5.10
@@ -1,2 +1,2 @@
-LINUX_VERSION-5.10 = .118
-LINUX_KERNEL_HASH-5.10.118 = c5e020f187a195ab0924fe7e71b8ab8fb92936ab47bc9d15e78daabddefe0b4b
+LINUX_VERSION-5.10 = .119
+LINUX_KERNEL_HASH-5.10.119 = 3e353e55a2b7074c905c7c517c6dae83eae440f98a12ffd471dc4df1d5172511
diff --git a/package/kernel/linux/modules/crypto.mk b/package/kernel/linux/modules/crypto.mk
index 6fd53f60806d..ecbae0eef10a 100644
--- a/package/kernel/linux/modules/crypto.mk
+++ b/package/kernel/linux/modules/crypto.mk
@@ -446,48 +446,6 @@ endef
 
 $(eval $(call KernelPackage,crypto-kpp))
 
-
-define KernelPackage/crypto-lib-blake2s
-  TITLE:=BLAKE2s hash function library
-  KCONFIG:=CONFIG_CRYPTO_LIB_BLAKE2S
-  HIDDEN:=1
-  FILES:=$(LINUX_DIR)/lib/crypto/libblake2s.ko
-  $(call AddDepends/crypto,+PACKAGE_kmod-crypto-hash:kmod-crypto-hash)
-endef
-
-define KernelPackage/crypto-lib-blake2s/config
-  imply PACKAGE_kmod-crypto-hash
-endef
-
-define KernelPackage/crypto-lib-blake2s/x86/64
-  KCONFIG+=CONFIG_CRYPTO_BLAKE2S_X86
-  FILES+=\
-	  $(LINUX_DIR)/lib/crypto/libblake2s-generic.ko \
-	  $(LINUX_DIR)/arch/x86/crypto/blake2s-x86_64.ko
-endef
-
-define KernelPackage/crypto-lib-blake2s/arm
-  KCONFIG+=CONFIG_CRYPTO_BLAKE2S_ARM
-  FILES+=\
-	  $(LINUX_DIR)/lib/crypto/libblake2s-generic.ko@lt5.12 \
-	  $(LINUX_DIR)/arch/arm/crypto/blake2s-arm.ko@ge5.12
-endef
-
-ifndef KernelPackage/crypto-lib-blake2s/$(CRYPTO_TARGET)
-  define KernelPackage/crypto-lib-blake2s/$(CRYPTO_TARGET)
-    KCONFIG+=CONFIG_CRYPTO_LIB_BLAKE2S_GENERIC
-    FILES+=$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
-  endef
-endif
-
-ifdef KernelPackage/crypto-lib-blake2s/$(ARCH)
-  KernelPackage/crypto-lib-blake2s/$(CRYPTO_TARGET)=\
-	  $(KernelPackage/crypto-lib-blake2s/$(ARCH))
-endif
-
-$(eval $(call KernelPackage,crypto-lib-blake2s))
-
-
 define KernelPackage/crypto-lib-chacha20
   TITLE:=ChaCha library interface
   KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA
diff --git a/package/kernel/linux/modules/netsupport.mk b/package/kernel/linux/modules/netsupport.mk
index 76187df50fb9..6fb3b57e126f 100644
--- a/package/kernel/linux/modules/netsupport.mk
+++ b/package/kernel/linux/modules/netsupport.mk
@@ -1325,7 +1325,6 @@ define KernelPackage/wireguard
   SUBMENU:=$(NETWORK_SUPPORT_MENU)
   TITLE:=WireGuard secure network tunnel
   DEPENDS:= \
-	  +kmod-crypto-lib-blake2s \
 	  +kmod-crypto-lib-chacha20poly1305 \
 	  +kmod-crypto-lib-curve25519 \
 	  +kmod-udptunnel4 \
diff --git a/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch b/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch
index f2f48d654325..f66f4fc4b62d 100644
--- a/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch
+++ b/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch
@@ -656,8 +656,8 @@
  #include <linux/types.h>
 +#include <linux/in6.h>
  
- u32 secure_ipv4_port_ephemeral(__be32 saddr, __be32 daddr, __be16 dport);
- u32 secure_ipv6_port_ephemeral(const __be32 *saddr, const __be32 *daddr,
+ u64 secure_ipv4_port_ephemeral(__be32 saddr, __be32 daddr, __be16 dport);
+ u64 secure_ipv6_port_ephemeral(const __be32 *saddr, const __be32 *daddr,
 --- a/include/uapi/linux/in.h
 +++ b/include/uapi/linux/in.h
 @@ -88,7 +88,7 @@ enum {
diff --git a/target/linux/generic/config-5.10 b/target/linux/generic/config-5.10
index c969c8e42278..516ec4fdf56a 100644
--- a/target/linux/generic/config-5.10
+++ b/target/linux/generic/config-5.10
@@ -1144,7 +1144,6 @@ CONFIG_CRYPTO_KPP=y
 CONFIG_CRYPTO_KPP2=y
 CONFIG_CRYPTO_LIB_AES=y
 CONFIG_CRYPTO_LIB_ARC4=y
-# CONFIG_CRYPTO_LIB_BLAKE2S is not set
 # CONFIG_CRYPTO_LIB_CHACHA is not set
 # CONFIG_CRYPTO_LIB_CHACHA20POLY1305 is not set
 # CONFIG_CRYPTO_LIB_CURVE25519 is not set
diff --git a/target/linux/generic/hack-5.10/261-lib-arc4-unhide.patch b/target/linux/generic/hack-5.10/261-lib-arc4-unhide.patch
index a7668acfadef..dfba811f31f2 100644
--- a/target/linux/generic/hack-5.10/261-lib-arc4-unhide.patch
+++ b/target/linux/generic/hack-5.10/261-lib-arc4-unhide.patch
@@ -4,7 +4,7 @@ from backports.
 
 --- a/lib/crypto/Kconfig
 +++ b/lib/crypto/Kconfig
-@@ -6,7 +6,7 @@ config CRYPTO_LIB_AES
+@@ -4,7 +4,7 @@ config CRYPTO_LIB_AES
  	tristate
  
  config CRYPTO_LIB_ARC4
@@ -12,4 +12,4 @@ from backports.
 +	tristate "ARC4 cipher library"
  
  config CRYPTO_ARCH_HAVE_LIB_BLAKE2S
- 	tristate
+ 	bool
diff --git a/target/linux/generic/pending-5.10/205-backtrace_module_info.patch b/target/linux/generic/pending-5.10/205-backtrace_module_info.patch
index f46877f254dd..f1f35c9923a6 100644
--- a/target/linux/generic/pending-5.10/205-backtrace_module_info.patch
+++ b/target/linux/generic/pending-5.10/205-backtrace_module_info.patch
@@ -11,7 +11,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 
 --- a/lib/vsprintf.c
 +++ b/lib/vsprintf.c
-@@ -983,8 +983,10 @@ char *symbol_string(char *buf, char *end
+@@ -985,8 +985,10 @@ char *symbol_string(char *buf, char *end
  		    struct printf_spec spec, const char *fmt)
  {
  	unsigned long value;
@@ -23,7 +23,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  #endif
  
  	if (fmt[1] == 'R')
-@@ -1001,8 +1003,14 @@ char *symbol_string(char *buf, char *end
+@@ -1003,8 +1005,14 @@ char *symbol_string(char *buf, char *end
  
  	return string_nocheck(buf, end, sym, spec);
  #else
diff --git a/target/linux/realtek/patches-5.10/008-5.17-watchdog-add-realtek-otto-watchdog-timer.patch b/target/linux/realtek/patches-5.10/008-5.17-watchdog-add-realtek-otto-watchdog-timer.patch
index cc5ee924669e..845441b4257f 100644
--- a/target/linux/realtek/patches-5.10/008-5.17-watchdog-add-realtek-otto-watchdog-timer.patch
+++ b/target/linux/realtek/patches-5.10/008-5.17-watchdog-add-realtek-otto-watchdog-timer.patch
@@ -32,7 +32,7 @@ Signed-off-by: Guenter Roeck <linux@roeck-us.net>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -14814,6 +14814,13 @@ S:	Maintained
+@@ -14816,6 +14816,13 @@ S:	Maintained
  F:	include/sound/rt*.h
  F:	sound/soc/codecs/rt*
  
